sw/device/tests/alert_handler_lpg_clkoff_test.c

```diff
@@ -0,0 +1,743 @@
+                                             /*he_en*/ false);
+
+  // Each test phase consists of ARRAYSIZE(kPeripherals) steps
+  // At every step, a specific peripheral is tested
+  size_t test_step_cnt = flash_ctrl_testutils_counter_get(kCounterTestSteps);
+  size_t test_phase = test_step_cnt / ARRAYSIZE(kPeripherals);
+  size_t peri_idx = test_step_cnt - (test_phase)*ARRAYSIZE(kPeripherals);
+
+  // At the beginning of the simulation (Verilator, VCS),
+  // the content of the flash might be all-zeros, and thus,
+  // the NVM counter's inital value might be 256.
+  // In that case, the NVM counter should be initialized to zero by filling
+  // its flash region with non-zero values.
+  // On FPGA, this code will be skipped
+  if (test_step_cnt == 256) {
+    flash_ctrl_testutils_counter_init_zero(&flash_ctrl, kCounterTestSteps);
+    test_step_cnt = flash_ctrl_testutils_counter_get(kCounterTestSteps);
+    test_phase = test_step_cnt / ARRAYSIZE(kPeripherals);
+    peri_idx = test_step_cnt - (test_phase)*ARRAYSIZE(kPeripherals);
+  }
+
+  /* TEST PHASE #0: NEGATIVE TESTING
+
+    The timeout value is set to 2
+    All of the alerts should trigger ping_timeout_alert
+    if they are not clock_gated
+    TEST: for_each_peripheral
+      1- Disable the clock
+      2- Enable the corresponding alerts
+      3- Wait long enough
+      4- Confirm that ping_timeout has not been triggerred
+      5- Reset the the device to test the next peripheral
+  */
+  while (test_step_cnt < 1 * ARRAYSIZE(kPeripherals)) {
+    // LOG_INFO("Test phase = %d, Periphereal idx = %d, test_step_cnt = %d",
+    // test_phase, peri_idx, test_step_cnt);
+    busy_spin_micros(rnd_wait_times[peri_idx]);
+
+    // Disable the clock of the peripheral
+    if (kPeripherals[peri_idx].is_hintable) {
+      CHECK_DIF_OK(dif_clkmgr_hintable_clock_set_hint(
+          &clkmgr, kPeripherals[peri_idx].clk_index, kDifToggleDisabled));
+    } else {
+      CHECK_DIF_OK(dif_clkmgr_gateable_clock_set_enabled(
+          &clkmgr, kPeripherals[peri_idx].clk_index, kDifToggleDisabled));
+    }
+
+    alert_handler_config_single_peripheral(&kPeripherals[peri_idx], 2);
+
+    // wait enough
+    if (kDeviceType == kDeviceFpgaCw310) {
+      // NUM_ALERTS*2*margin_of_safety*(2**DW)*(1/kClockFreqPeripheralHz)
+      // (2**6)*2*4*(2**16)*(400ns) = 12.8s
+      busy_spin_micros(1000 * 12800);
+    } else if (kDeviceType == kDeviceSimDV) {
+      // NUM_ALERTS*2*margin_of_safety*(2**DW)*(1/kClockFreqPeripheralHz)
+      // (2**6)*2*4*(2**3)*(40ns) = 160us
+      busy_spin_micros(160);
+    }
+
+    // Check the ping_timeout alert status
+    dif_alert_handler_local_alert_t check_local_alert =
+        kDifAlertHandlerLocalAlertAlertPingFail;
+    CHECK_DIF_OK(dif_alert_handler_local_alert_is_cause(
+        &alert_handler, check_local_alert, &is_cause));
+    if (is_cause) {
+      LOG_INFO("png_timeout for peripheral[%d] is 1", peri_idx);
```

Comment:
nit: `png_timeout` -> `ping_timeout`

Reply:
You can probably combine this error message with `CHECK` below, since it supports print arguments.

Reply:
Good suggestion, I combined them.